app-text/sigil: version bump to 0.9.13

arthurzam opened the pull request:
Please note that the package needs a new maintainer, based on what [this page](https://packages.gentoo.org/packages/app-text/sigil) shows, and I will gladly take over as a  Proxy Maintainer - but I don't know the procedures.

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
7 should work here, please use the latest EAPI available.

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,82 @@
+
+inherit cmake-utils python-single-r1
+
+my_pn="Sigil"
```
Global variables are written with upcase lettering, you should fix this and update the variable used through this ebuild.

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,82 @@
+
+RDEPEND="
+	${PYTHON_DEPS}
+	app-text/hunspell
```
This most likely needs a subslot bind. `:=`

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,82 @@
+RDEPEND="
+	${PYTHON_DEPS}
+	app-text/hunspell
+	dev-libs/boost[threads]
```
This definitely needs a subslot bind, `:=`

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,82 @@
+	${PYTHON_DEPS}
+	app-text/hunspell
+	dev-libs/boost[threads]
+	dev-libs/libpcre[pcre16]
```
Specify slot here please.

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,82 @@
+	sys-libs/zlib[minizip]
+"
+DEPEND="${RDEPEND}
+	>=sys-devel/gcc-4.8
```
This is not the correct way to force `gcc >=4.8`, please use toolchain-funcs eclass to do that. 

Something like this should work, 
```
pkg_pretend() {
	if tc-is-gcc && [[ $(gcc-version) -lt 4.8 ]] ; then
		die "Active GCC needs to be newer than 4.8"
	fi
}
```
but please test it.

juippis commented:
> I will gladly take over as a Proxy Maintainer - but I don't know the procedures.

Edit the `metadata.xml` file, adding yourself and the proxy-maint project there. See example from already proxy-maintained packages :)

You need a bugzilla account with that email you add to the metadata.

arthurzam commented:
@juippis Is it better to force push or create a new commit in this pull request?

juippis commented:
If you have the knowledge to rebase and force-push, do that please. 

So fix the ebuild, make a new commit, rebase and squash and that commit into your first commit so there is only one commit with version bump, then make a new commit updating metadata.xml file. 
So in the end there are 2 commits, one with version bump, one with metadata.xml update.

arthurzam commented:
@juippis Big thank you for your help!

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2018 Gentoo Foundation
```
You should use repoman to fix this copyright header automatically for you, but for now, update to 
`# Copyright 1999-2019 Gentoo Authors`

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,87 @@
+	sys-libs/zlib[minizip]
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
With EAPI-7 this is a BDEPEND. 
`BDEPEND="virtual/pkgconfig`

I have a feeling that unzip below also is a BDEPEND, but I do not know for sure. If unzip is executed during the build to extract some .zip file only, then it's a BDEPEND.

juippis commented on app-text/sigil/sigil-0.9.12.ebuild:
```diff
@@ -0,0 +1,87 @@
+DOCS=( ChangeLog.txt README.md )
+
+pkg_pretend() {
+	if tc-is-gcc && [ $(gcc-version) -lt 4.8 ] ; then
```
Please use double brackets, `[[ ]]` it's less error-prone.

juippis commented:
Sorry I missed few things in the initial round, but overall it's looking good!

arthurzam commented:
> Sorry I missed few things in the initial round, but overall it's looking good!
> 
> EDIT: The error above seems to be from dev-python/cssutils not being updated for python3.7 yet. You can update this ebuild to be compatible with python3.6 max, or investigate, test and update cssutils to support python3.7 before this gets merged. You can put the python3.7 bump to this PR, or make a new PR for that. You can also wait for this to get merged with python3.6, then make a new PR to update cssutils, and after that is merged, make a new PR to add python3.7 to this package. That's probably the best way.

Thank for your comments.
I put a bug in [bugzilla](https://bugs.gentoo.org/679494). Is it better to put a pull request?
About the version check, after looking in `dev-util/electron` ebuild, I found the `ver_test` from EAPI 7, which I think is better.

juippis commented:
> I put a bug in [bugzilla](https://bugs.gentoo.org/679494). Is it better to put a pull request?

It's good, the python project should be pretty active both in bugzilla and Github. Whatever is best for you, as long as you can track the progress.

> About the version check, after looking in `dev-util/electron` ebuild, I found the `ver_test` from EAPI 7, which I think is better.

Yes it should work too. You can choose the implementation yourself, whatever you find easier and more to your liking. Bear in mind, using ver_test still requires you to inherit toolchain-funcs for gcc-version.

arthurzam commented:
The problem with `dev-python/cssutils` python3.7 support was fixed.

Upstream version was updated to `0.9.13` - updated this PR.
Added missing call for `xdg_desktop_database_update` from `xdg-utils.eclass`
Moved `dev-qt/linguist-tools:5` into BDEPEND, as done in other Qt projects.
Changed `SRC_URI` to use `.tar.gz` - removed `app-arch/unzip` dependency.

arthurzam commented:
@gentoo/proxy-maint Could someone check and maybe merge it?

mgorny commented on app-text/sigil/files/sigil-0.9.13-proper-gumbo-install.patch:
```diff
@@ -0,0 +1,24 @@
+ 
+ if( UNIX AND NOT APPLE )
+ 	set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99" )
++    install( TARGETS sigilgumbo DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR}/sigil/ )
```
Why are you moving it? Also, is it building a bundled gumbo instead of using dev-libs/gumbo? Also, mismatched indentation.

arthurzam replied:
The gumbo is patched with patches no accepted by upstream (found in wiki of Sigil). About the install patch, changed.

mgorny commented on app-text/sigil/files/sigil-0.9.13-proper-gumbo-install.patch:
```diff
@@ -0,0 +1,24 @@
+-    install( TARGETS sigil DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/sigil/ )
++    install( TARGETS sigil DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR}/sigil/ )
+     install( PROGRAMS ${LINUX_LAUNCH_INSTALL_SCRIPT_CONFIGURED} DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/ RENAME ${PROJECT_NAME} )
+-    if( NOT USE_SYSTEM_LIBS OR NOT HUNSPELL_FOUND )
```
Do you really need to remove this?

mgorny commented on app-text/sigil/files/sigil-0.9.13-proper-gumbo-install.patch:
```diff
@@ -0,0 +1,24 @@
+ 	set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99" )
++    install( TARGETS sigilgumbo DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR}/sigil/ )
+ endif()
+--- a/src/CMakeLists.txt
```
Why not use GNUInstallDirs instead? Then you'd have a patch you could actually reasonably submit upstream.

mgorny commented on app-text/sigil/sigil-0.9.13.ebuild:
```diff
@@ -0,0 +1,93 @@
+)
+DOCS=( ChangeLog.txt README.md )
+
+pkg_pretend() {
```
I dare say that nobody should be using gcc < 5.4 these days (and probably even newer), due to profiles.

arthurzam replied:
Removed.

mgorny commented on app-text/sigil/sigil-0.9.13.ebuild:
```diff
@@ -0,0 +1,93 @@
+}
+
+pkg_postinst() {
+	elog "From Sigil's release notes:"
```
This sounds like case for readme.gentoo-r1 eclass.

arthurzam replied:
Done.

arthurzam commented:
- Removed bundled in package `mathjax` (now depends on `dev-libs/mathjax`)
- Removed bundled `hunspell` dictionaries
- Added `readme.gentoo-r1`
- Changed the install patch to use `GNUInstallDirs`. Spoke with maintainer about including the patch upstream, they are against because they are afraid to break other distribution's packages. But they said they are planning to integrate a way to select at least `lib -> lib64` (which is what we need for this package).

Another question:
The package includes some python modules for sigil. How (and if) should I python_optimize (I saw it in some other ebuilds but haven't understood how)?

mgorny commented:
You can find eclass documentation either inside the files, in eclass-manpages package or in devmanual.

mgorny commented:
Oh, and python eclasses are also documented in detail on wiki, in Project:Python.

arthurzam commented:
@mgorny I hope I have done it right (no that sure about `python_optimize` - mainly about the passed argument).
Also fixed license (internal gumbo is licensed under `Apache-2`). Is it correct?

mgorny commented on app-text/sigil/files/sigil-0.9.13-fix-install.patch:
```diff
@@ -0,0 +1,62 @@
+    SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+ 
+-   SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib/sigil")
++   SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}/sigil")
```
I'm sorry but now that I see the patch, I don't think libdir is really necessary here. As long as it installs only big 'sigil' directory, /usr/lib is preferable as it keeps the path constant.

arthurzam replied:
Done.
Now the patch can be removed.

mgorny commented on app-text/sigil/sigil-0.9.13.ebuild:
```diff
@@ -0,0 +1,93 @@
+
+inherit cmake-utils python-single-r1 readme.gentoo-r1 xdg-utils
+
+MY_PN="Sigil"
```
Make it into MY_P (add the version), and rename the tarball to MY_P rather than P. It is surprising when the directory inside it is extracted to a different name than tarball itself.

mgorny commented on app-text/sigil/sigil-0.9.13.ebuild:
```diff
@@ -0,0 +1,93 @@
+
+DESCRIPTION="Sigil is a multi-platform WYSIWYG ebook editor for ePub format"
+HOMEPAGE="https://sigil-ebook.com/"
+SRC_URI="https://github.com/Sigil-Ebook/${MY_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
```
Please inline the MY_PN in first part of GitHub link, so that people can easily copy-paste it into browser.

mgorny commented on app-text/sigil/sigil-0.9.13.ebuild:
```diff
@@ -0,0 +1,93 @@
+src_install() {
+	cmake-utils_src_install
+	python_optimize "${ED}"/usr/share/sigil/
+	DISABLE_AUTOFORMATTING=true	readme.gentoo_create_doc
```
Don't use tab in middle of the line.

arthurzam commented:
@mgorny Done.
Never thought about the download name and extracted name. Everyday you learn something new and interesting :)

mgorny commented:
Thanks. Please give me some time to build qtwebkit.

mgorny commented:
You seem to be missing some deps still:

```
Traceback (most recent call last):

  File "/usr/share/sigil/python3lib/xmlprocessor.py", line 6, in <module>
    from sigil_bs4.builder._lxml import LXMLTreeBuilderForXML

  File "/usr/share/sigil/plugin_launchers/python/sigil_bs4/builder/_lxml.py", line 26, in <module>
    from lxml import etree

ImportError: No module named 'lxml'
```

arthurzam commented:
> You seem to be missing some deps still:
> 
> ```
> Traceback (most recent call last):
> 
>   File "/usr/share/sigil/python3lib/xmlprocessor.py", line 6, in <module>
>     from sigil_bs4.builder._lxml import LXMLTreeBuilderForXML
> 
>   File "/usr/share/sigil/plugin_launchers/python/sigil_bs4/builder/_lxml.py", line 26, in <module>
>     from lxml import etree
> 
> ImportError: No module named 'lxml'
> ```

Interesting, as `dev-python/lxml[${PYTHON_USEDEP}]` is a dependency.

mgorny commented:
Ok, there's something really wrong going on here:

```
 0x0000000000000001 (NEEDED)             Shared library: [libpython3.4m.so.1.0]
```

So it's ignoring PYTHON_SINGLE_TARGET completely.

arthurzam commented:
@mgorny Well, that is really weird as I haven't touched anything about the python usage. I guess it came from the previous ebuilds. I think I know where to check and fix it. Will take some time.

arthurzam commented:
@mgorny I hope I fixed (but I ask you to recheck it).
I added a call to `python_fix_shebang`.
Somehow until now the check in `cmake` worked with some python wrapper magic (well I have no idea how but it found somehow the correct version). But to be sure I now explicitly pass as an argument the correct python executable.

mgorny commented:
There's still something wrong with it:

```
-- Found PythonInterp: /usr/bin/python3.7 (found suitable version "3.7.3", minimum required is "3.4") 
-- Found PythonLibs: /usr/lib64/libpython3.4m.so (found suitable version "3.4.10", minimum required is "3.4") 
```

[app-text:sigil-0.9.13:20190429-132026.log](https://github.com/gentoo/gentoo/files/3127544/app-text.sigil-0.9.13.20190429-132026.log)

I don't see anything useful in CMakeOutput.txt.

arthurzam commented:
@mgorny First of all, sorry for all this trouble. I sincerely am apologizing for wasting your time.
I found the culprit! In [this](https://github.com/Sigil-Ebook/Sigil/blob/master/CMakeLists.txt#L96) line, if python3.4 is installed it wins the previous configuration of using `PYTHON_EXECUTABLE`. 
Haven't found this bug previously as I don't have installed python3.4 and I tries various changes with 3.5 and 3.6 (main system uses 3.7). I managed to reproduce your report by installing python3.4.
Fixed with passing all needed python configurations.
I sorry again to ask it, but please check it again.

mgorny commented:
No need to apologize. Cool you found the culprit. FWICS it built fine this time and runs, so I'll merge it. Thanks!

mgorny commented:
For the future, I'd suggest:

```
set directory=~/tmp,/var/tmp
```

This will stop vim from making .swp files in current directory, and you from committing them ;-).

eli-schwartz commented:
> Changed the install patch to use `GNUInstallDirs`. Spoke with maintainer about including the patch upstream, they are against because they are afraid to break other distribution's packages. But they said they are planning to integrate a way to select at least `lib -> lib64` (which is what we need for this package).

(This is now merged to master, with a suitable changelog message.)

BTW a bunch of these modules have never been a hard dep (in my Arch packaging I use an optional dependency for many of the python modules) but cssutils has been essentially dropped in favor of css-parser. I've confirmed with Doug and Kevin that for my distro packaging purposes it's fine to `rm -r src/Resource_Files/plugin_launchers/python/css_parser/` and depend on the python-css-parser package instead -- you may want to do the same.
